spec/models/user_spec.rb

```diff
@@ -51,6 +51,47 @@
     expect(new_user.errors.messages[:email]).to include('has already been taken')
   end
 
+  describe '#destroy' do
```

Comment:
These tests are probably not necessary because they're testing the default behavior of Rails, including the "delete cascading" - if it was me, I would remove these tests (and the restrict_with_errors as I mentioned above) and assume Rails will handle this correctly.